### ISSUE TITLE
Perplexed by commands not working in manual mode

### ISSUE THREAD

**Issue by jryans** [modernweb-dev/web#1576]
I am a bit confused by commands are restricted to manual mode... Is it just a technical limitation because you need a session ID to manage the WebSocket connection...? I realise some of the default commands perhaps only work in a controlled browser, but that's not (necessarily) the case for custom commands.

I have custom commands I'd like to offer in manual mode as well, and at the moment, it looks like I'd need to start my own WebSocket server to get around this limitation, which feels a bit silly as I'll be duplicating a lot of what commands offer.

**Comment by LarsDenBakker** [comment]
Commands need to access the puppeteer/webdriver instance controlling the webpage. In manual mode, this isn't the case as you're running a custom a browser.

**Comment by jryans** [comment]
Right, that's true for the _default_ commands... but there's also support for custom commands, which could do anything at all. They might not need a controlled browser (which happens to be true for my case), so it's unfortunate that the entire commands infrastructure requires a controlled browser just because the default ones need it.

Perhaps the controlled browser requirement should happen on a per-command level...?

**Comment by LarsDenBakker** [comment]
The commands infrastructure exists for code that requires access to puppeteer/webdriver etc. If this isn't necessary, they're more like API endpoints which can be implemented as a plugin or middleware.

**Comment by jryans** [comment]
Okay, indeed I just realised I can achieve what I want via a [WebSocket plugin](https://modern-web.dev/docs/dev-server/writing-plugins/web-sockets/), thanks.

Maybe it's a documentation problem then...? When reading about [commands](https://modern-web.dev/docs/test-runner/commands/), is says:

> Commands for executing code server-side during your tests in the browser. To control the browser page, access the file system, or execute NodeJS libraries.

I don't need a controlled browser for 2 of those 3 examples... Maybe the purpose of commands has changed since that was written?
